### ISSUE TITLE
What is Killeding

### ISSUE THREAD

**Issue by irbull** [git-lfs/git-lfs#5518]
**Describe the issue**
I have been trying to `git clone` a git repository with large files and it reports that it's `Killeding content`. Things are failing, and I'm not sure if this is related.

**System environment**
The original repository was created on a Linux/Ubuntu image and pushed to Hugging Face. I'm trying to pull the repository from HuggingFace using another Linux/Ubuntu image.

**Output of `git lfs env`**

```
git version 2.39.2

Endpoint=https://hf.co/... (auth=none)
  SSH=git@hf.co:...
LocalWorkingDir=/root/...
LocalGitDir=/root/.../.git
LocalGitStorageDir=/root/.../.git
LocalMediaDir=/root/.../.git/lfs/objects
LocalReferenceDirs=
TempDir=/root/.../.git/lfs/tmp
ConcurrentTransfers=8
TusTransfers=false
BasicTransfersOnly=false
SkipDownloadErrors=false
FetchRecentAlways=false
FetchRecentRefsDays=7
FetchRecentCommitsDays=0
FetchRecentRefsIncludeRemotes=true
PruneOffsetDays=3
PruneVerifyRemoteAlways=false
PruneRemoteName=origin
LfsStorageDir=/root/.../.git/lfs
AccessDownload=none
AccessUpload=none
DownloadTransfers=basic,lfs-standalone-file,ssh
UploadTransfers=basic,lfs-standalone-file,ssh
GIT_EXEC_PATH=/usr/lib/git-core
git config filter.lfs.process = "git-lfs filter-process"
git config filter.lfs.smudge = "git-lfs smudge -- %f"
git config filter.lfs.clean = "git-lfs clean -- %f"
```

**Additional context**
I tried to perform a GitClone and it finished:

```
Cloning into '...'...
remote: Enumerating objects: 28, done.
remote: Counting objects: 100% (25/25), done.
remote: Compressing objects: 100% (25/25), done.
remote: Total 28 (delta 5), reused 0 (delta 0), pack-reused 3
Receiving objects: 100% (28/28), 5.95 KiB | 5.95 MiB/s, done.
Resolving deltas: 100% (5/5), done.
Updating files: 100% (18/18), done.
Killeding content: 100% (6/6), 4.85 GiB | 2.07 MiB/s
```
Each file in the repository was marked as deleted in the Git tree and some files are missing.

I tried a `git reset --hard HEAD` and it completed with:

```
Killedng files:  44% (8/18)
```

Again, all files were marked as deleted and some files are missing.

**Comment by irbull** [comment]
🤔 Hum, I think this might be a bad terminal messing the output.

It very well could be `Filtering content` and then something is killing the process and `Killed` is being written without a newline. May explain why `Killeding` is not something I've ever seen before. I'm thinking it could be OOM, but I'm not sure.

But that leads to the next question, how much memory does git-lfs use? There is certainly many GB (~10GB) available, but I'm trying to checkout a repository with a 13GB file in it. Does it require the entire file to be written to memory?

**Comment by bk2204** [comment]
Hey,

This is indeed `Filtering content` with `Killed` written over it by your shell.  My guess is that this is Git being killed, because typically if Git LFS, which is invoked by Git, were killed, your shell wouldn't print any messages.

Git LFS doesn't require that the file be read entirely into memory, but in some cases Git may.  You can try to use `git lfs pull` to check out the Git LFS files, and if you want to see if Git's interaction with Git LFS is the cause of this, then you can run `git lfs install --skip-smudge` to ask Git LFS not to smudge files into the working tree by default.  Then you can run `git lfs pull` if you need to check out the LFS files, and in some cases that fixes the memory problems.

However, if that doesn't work, I'd poke more into Git and see why it's using all that memory.

**Comment by irbull** [comment]
Yep, it was exactly this. I was running it in a container (on k8s) and I had a memory limit of 10Gb set. When I increased it to 14Gb it succeeded. I assumed that `git clone` was just a transport layer, so even if it had to transfer 100Gb file, it wouldn't need a lot of memory but I've never looked at the implementation; it's probably a lot more complicated than that.

This doesn't really affect me personally, as I can just give the container more memory. If you feel it shouldn't need that much ram to clone a large repo we could investigate this with Git.

In them meantime I'll close this as there really isn't anything to do. As an aside, while googling `Killeding` I did find another article. It was a blog that someone wrote that included some output that had `Killeding content` in it. I guess I'm not the first person to hit this. Hopefully the next time it happens to someone they will find this report instead. Thanks for your help Brian.
